PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java

```diff
@@ -9,107 +9,109 @@
         final List<T> items = loadAppServiceModels();
-        if (configModel != null && configModel.isNewCreateResource()) {
-            final boolean exist = items.stream().anyMatch(item -> AppServiceComboBoxModel.isSameApp(item, configModel));
+        if (isDraftResource(configModel)) {
+            final boolean exist = items.stream().anyMatch(item -> AppServiceConfig.isSameApp(item, configModel));
             if (!exist) {
                 items.add(configModel);
             }
         }
         return items;
     }
 
+    @Override
+    public T getValue() {
+        if (value instanceof ItemReference && ((ItemReference<?>) value).is(configModel)) {
+            return configModel;
+        }
+        return super.getValue();
+    }
+
     protected abstract List<T> loadAppServiceModels() throws Exception;
 
-    @Nullable
+    @Nonnull
     @Override
     protected ExtendableTextComponent.Extension getExtension() {
-        return ExtendableTextComponent.Extension.create(
-            AllIcons.General.Add, "Create", this::createResource);
+        return ExtendableTextComponent.Extension.create(AllIcons.General.Add, "Create", this::createResource);
     }
 
     @Override
     protected String getItemText(final Object item) {
-        if (item instanceof AppServiceComboBoxModel) {
-            final AppServiceComboBoxModel selectedItem = (AppServiceComboBoxModel) item;
-            return selectedItem.isNewCreateResource() ?
-                String.format("(New) %s", selectedItem.getAppName()) : selectedItem.getAppName();
+        if (item instanceof AppServiceConfig) {
+            final AppServiceConfig selectedItem = (AppServiceConfig) item;
+            return isDraftResource(selectedItem) ? String.format("(New) %s", selectedItem.getName()) : selectedItem.getName();
         } else {
             return Objects.toString(item, StringUtils.EMPTY);
         }
     }
 
-    protected boolean isJavaAppService(IAppService appService) {
-        try {
-            return Optional.ofNullable(appService.getRuntime()).map(Runtime::getJavaVersion)
-                    .map(javaVersion -> !Objects.equals(javaVersion, JavaVersion.OFF))
-                    .orElse(false);
-        } catch (final RuntimeException e) {
-            // app service may have been removed while parsing, return false in this case
-            return false;
-        }
-    }
-
     protected abstract void createResource();
 
     public static class AppComboBoxRender extends SimpleListCellRenderer {
 
         @Override
-        public void customize(JList list, Object value, int index, boolean b, boolean b1) {
-            if (value instanceof AppServiceComboBoxModel) {
-                final AppServiceComboBoxModel app = (AppServiceComboBoxModel) value;
+        public void customize(JList list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
+            if (value instanceof AppServiceConfig) {
```

Comment:
is it possible not a `AppServiceConfig`?

Reply:
it should always be AppServiceConfig but may be null here